src/server/game/Entities/Player/Player.cpp

```diff
@@ -1841,6 +1841,9 @@ void Player::Regenerate(Powers power)
         }
     }
 
+    if (IsPlayer() && ToPlayer()->GetCommandStatus(CHEAT_POWER))
```

Comment:
I would say that `this` in a function belonging to `Player` class is always a `Player`